chocolatey-diff/Public/Get-FullPackageInfo.ps1

```diff
@@ -19,24 +19,27 @@ function Get-FullPackageInfo {
         [string] $PackageId
     )
 
-    $pkgInfo = @{
-        Id       = $PackageId
-        Url      = "$(Get-ChocolateyPackageGalleryUrl)/${$PackageId}"
-        Versions = @()
-    }
+    if (!(Get-Variable -Name pkgInfo -Scope Script -ErrorAction SilentlyContinue)) {
+
+        $script:pkgInfo = @{
+            Id       = $PackageId
+            Url      = "$(Get-ChocolateyPackageGalleryUrl)/${$PackageId}"
+            Versions = @()
+        }
 
-    $url = "$(Get-ChocolateyRepositoryUrl)/package-versions/${PackageId}?includePreRelease=true"
-    [array]$jsonVersions = Invoke-RestMethod -Uri $url -UseBasicParsing | Select-Object -Unique
-    $apiVersions = $jsonVersions | ForEach-Object { [SemanticVersion]::create($_) } | Sort-Object -Property VersionOnly, Tag -Descending | Select-Object -Unique
+        $url = "$(Get-ChocolateyRepositoryUrl)/package-versions/${PackageId}?includePreRelease=true"
+        [array]$jsonVersions = Invoke-RestMethod -Uri $url -UseBasicParsing | Select-Object -Unique
+        $apiVersions = $jsonVersions | ForEach-Object { [SemanticVersion]::create($_) } | Sort-Object -Property VersionOnly, Tag -Descending | Select-Object -Unique
 
-    foreach ($v in $apiVersions) {
-        $vInfo = Get-VersionData -packageName $PackageId -version $([SemanticVersion]::create($v))
-        $pkgInfo.Versions += @{
-            Version = $vInfo.version
-            Listed  = $vInfo.listed
-            Status  = $vInfo.status
+        foreach ($v in $apiVersions) {
+            $vInfo = Get-VersionData -packageName $PackageId -version $([SemanticVersion]::create($v))
```

Comment:
Why recreate the version here? You already created it just above (L32).

Reply:
Changed